.github/workflows/images.yaml

```diff
@@ -0,0 +1,51 @@
+# NUC image creation and push to DockerHub
+name: NUC Image Push
+
+# Controls when the action will run.
+on:
+  # Triggers the workflow on push events for the main branch
+  push:
+    branches: [main]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # Build and push all images
+  build_and_push:
+    name: "Build and Push Docker Images"
+
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        platform: ["generic", "nuc8i7beh", nuc7i7bnh]
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      # Setup docker buildx
+      - name: 🐳 Set up docker buildx 🐳
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+      # Build the docker image
+      - name: 🐳 Build the docker image 🐳
+        uses: docker/build-push-action@v2
+        with:
+          pull: true
+          tags: "nubots/nubots:${{ matrix.platform }}"
```

Comment:
Similar here, It might be nice to read in all references to our docker registry from repository environment/secrets instead of having it hardcoded in the workflow. This would be more relevent for when we add a RoboCup image workflow as we could easily change where the image is pushed by changing the repository environment variable.

Reply:
I would consider the RoboCup image (if you mean automatically upload the current RoboCup image to the competition during the competition time) to be an entirely separate workflow from this. Lots of very different steps.

That said, I think if you are in a contest it would be very cool to have a branch that automatically uploads the image to the competition. That way you will never again forget to push up the contest, it'll always be whatever is on that branch

Reply:
Thats the plan :+1: , and yeah definetly a separate workflow. I guess its better for changes to the registry address to go through a review process rather than just changing a repo variable.